azure-mgmt-dns/src/main/java/com/microsoft/azure/management/dns/implementation/DnsZoneImpl.java

```diff
@@ -414,31 +479,67 @@ private void initRecordSets() {
+                        return Observable.just((DnsRecordSet) new ARecordSetImpl(inner.name(), self, inner));
                     case AAAA:
-                        return Observable.just((DnsRecordSet) new AaaaRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new AaaaRecordSetImpl(inner.name(), self, inner));
+                    case CAA:
+                        return Observable.just((DnsRecordSet) new CaaRecordSetImpl(inner.name(), self, inner));
                     case CNAME:
-                        return Observable.just((DnsRecordSet) new CNameRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new CNameRecordSetImpl(inner.name(), self, inner));
                     case MX:
-                        return Observable.just((DnsRecordSet) new MXRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new MXRecordSetImpl(inner.name(), self, inner));
                     case NS:
-                        return Observable.just((DnsRecordSet) new NSRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new NSRecordSetImpl(inner.name(), self, inner));
                     case PTR:
-                        return Observable.just((DnsRecordSet) new PtrRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new PtrRecordSetImpl(inner.name(), self, inner));
                     case SOA:
-                        return Observable.just((DnsRecordSet) new SoaRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new SoaRecordSetImpl(inner.name(), self, inner));
                     case SRV:
-                        return Observable.just((DnsRecordSet) new SrvRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new SrvRecordSetImpl(inner.name(), self, inner));
                     case TXT:
-                        return Observable.just((DnsRecordSet) new TxtRecordSetImpl(self, inner));
+                        return Observable.just((DnsRecordSet) new TxtRecordSetImpl(inner.name(), self, inner));
                     default:
                         return Observable.just(recordSet);
                 }
             }
         };
         return converter.convert(manager().inner().recordSets().listByDnsZone(this.resourceGroupName(), this.name(), pageSize, recordSetSuffix));
     }
+
+    @Override
+    public DnsZoneImpl withPublicAccess() {
+        this.inner().withZoneType(ZoneType.PUBLIC);
+        this.inner().withRegistrationVirtualNetworks(null);
+        this.inner().withResolutionVirtualNetworks(null);
+        return this;
+    }
+
+    @Override
+    public DnsZoneImpl withPrivateAccess() {
+        this.inner().withZoneType(ZoneType.PRIVATE);
+        this.inner().withRegistrationVirtualNetworks(null);
+        this.inner().withResolutionVirtualNetworks(null);
```

Comment:
Is it safe to reset the reg and resolution networks? let say user is updating an existing private zone, if he choose `withPrivateAccess` then that will clear all existing reg and resolution networks without user knowledge.